lc-0141-linked-list-cycle/src/test/java/solution/SolutionTest.java

```diff
@@ -0,0 +1,61 @@
+package solution;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SolutionTest {
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "[3,2,0,-4]; 0; true",
+            "[3,2,0,-4]; -1; false",
+            "[3,2,0,-4]; 2; true",
+            "[1,2]; 0; true",
+            "[]; -1; false"
```

Comment:
We've hit the limits of CsvSource and this is no longer readable. Two issues with this:

- The columns are not labeled. Without reading the test code, it's very hard to infer that the second number is the node which has a cycle.

- It's not immediately clear that -1 means no cycle. And it's not immediately clear that -1 is not special - any negative number means "no cycle". A better type would be Optional<Integer>. It would hold nothing when there is no cycle, or the cycle position otherwise. But as I said above, we're hitting the limits of CsvSource.

Reply:
Agreed. The inputs are based on the problem statement inputs. However, going forward, will separate testing different aspects of the problem (for example values vs objects). Cramming everything in one big tests creates potential for bugs and errors of its own.

Resolving this without updating code since we have many more upcoming problems where this can be addressed.